common/src/main/java/org/opensearch/ml/common/httpclient/MLHttpClientFactory.java

```diff
@@ -83,35 +78,53 @@ private static boolean hasPrivateIpAddress(InetAddress[] ipAddress) {
                 if (bytes.length != 4) {
                     return true;
                 } else {
-                    int firstOctets = bytes[0] & 0xff;
-                    int firstInOctal = parseWithOctal(String.valueOf(firstOctets));
-                    int firstInHex = Integer.parseInt(String.valueOf(firstOctets), 16);
-                    if (firstInOctal == 127 || firstInHex == 127) {
-                        return bytes[1] == 0 && bytes[2] == 0 && bytes[3] == 1;
-                    } else if (firstInOctal == 10 || firstInHex == 10) {
+                    // case 127.0.0.1
+                    boolean isLocalHost = eqCheckValue(bytes[0], 127)
+                        && eqCheckValue(bytes[1], 0)
+                        && eqCheckValue(bytes[2], 0)
+                        && eqCheckValue(bytes[3], 1);
+                    if (isLocalHost) {
                         return true;
-                    } else if (firstInOctal == 172 || firstInHex == 172) {
-                        int secondOctets = bytes[1] & 0xff;
-                        int secondInOctal = parseWithOctal(String.valueOf(secondOctets));
-                        int secondInHex = Integer.parseInt(String.valueOf(secondOctets), 16);
-                        return (secondInOctal >= 16 && secondInOctal <= 32) || (secondInHex >= 16 && secondInHex <= 32);
-                    } else if (firstInOctal == 192 || firstInHex == 192) {
-                        int secondOctets = bytes[1] & 0xff;
-                        int secondInOctal = parseWithOctal(String.valueOf(secondOctets));
-                        int secondInHex = Integer.parseInt(String.valueOf(secondOctets), 16);
-                        return secondInOctal == 168 || secondInHex == 168;
                     }
+                    // case 10.x.x.x
+                    return eqCheckValue(bytes[0], 10) ||
+                    // case 172.16.x.x - 172.31.x.x
+                        (eqCheckValue(bytes[0], 172) && rangeCheckValue(bytes[1])) ||
+                        // case 192.168.x.x
+                        (eqCheckValue(bytes[0], 192) && eqCheckValue(bytes[1], 168)) ||
+                        // case 169.254.x.x
+                        (eqCheckValue(bytes[0], 169) && eqCheckValue(bytes[1], 254));
                 }
             }
         }
         return Arrays.stream(ipAddress).anyMatch(x -> x.isSiteLocalAddress() || x.isLoopbackAddress() || x.isAnyLocalAddress());
     }
 
-    private static int parseWithOctal(String input) {
+    private static boolean eqCheckValue(byte input, int targetValue) {
+        int original = input & 0xff;
+        return original == targetValue || parseWithRadix(original, 8) == targetValue || parseWithRadix(original, 16) == targetValue;
+    }
+
```

Comment:
Performance concern: This method attempts octal and hex parsing for every byte comparison, which will frequently throw NumberFormatException for common IP octets (192, 168, 127, etc.) since they contain invalid octal digits. Consider using simple decimal comparison since IP addresses are standardized as decimal.

Reply:
IP addresses are not simple standardized decimals if you check the UTs

Reply:
Tested, these parsing are indeed not necessary as when `byte[] bytes = ip.getAddress();` runs, it already parsed the strange ip address to correct numbers, so those parsing code are not get executed.